crates/type-system/src/ontology/uri/mod.rs

```diff
@@ -87,9 +89,14 @@ impl FromStr for VersionedUri {
     type Err = ParseVersionedUriError;
 
     fn from_str(uri: &str) -> Result<Self, ParseVersionedUriError> {
-        let (base_uri, version) = uri.rsplit_once("/v/").ok_or(ParseVersionedUriError)?;
-
         // TODO: better error handling
+        lazy_static! {
+            static ref RE: Regex = Regex::new(r#"(.*/)v/(\d)*"#).expect("Regex failed to compile");
```

Comment:
I think this Regex should be changed to the following
```suggestion
            static ref RE: Regex = Regex::new(r#"(.+/)v/(\d+)"#).expect("Regex failed to compile");
```
otherwise the capture group number 2 will capture the right-most number, and the regex would match against `/v/`.

Reply:
As discussed offline, we needed more modifications